qcfractal/postgres_harness.py

```diff
@@ -338,6 +343,54 @@ def init_database(self) -> None:
             self.logger(ret)
             raise ValueError("\nFailed to Stamp the database with current version.\n")
 
+    def backup_database(self, filename: Optional[str] = None) -> None:
+
+        # Reasonable check here
+        self._check_psql()
+
+        if filename is None:
+            filename = os.path.realpath(f"{self.config.database.database_name}.bak")
+
+        # fmt: off
+        cmds = [
+            shutil.which("pg_dump"),
+            "-p", str(self.config.database.port),
+            "-d", self.config.database.database_name,
+            "-Fc", # Custom postgres format, fast
+            "--file", filename
```

Comment:
Consider doing this in a faux-atomic way to avoid writing incomplete backups during failures. For example, write the backup to a temporary file, then move it to `filename` when done.

Reply:
On the fence on this one. I like it because the user can see that there is something actually happening, but in rare cases in can result in incomplete files.

Reply:
I have left the current version for now.

Reply:
You could write a file in the same directory called `f"{filename}.in_progress"` and then move it when done.

Reply:
Good point, added.